spring-cloud-generator/scripts/generate-steps.sh

```diff
@@ -27,14 +27,22 @@ function compute_monorepo_version() {
 }
 
 # Generate list of in-scope libraries to use (as static data source to generate modules from)
-# Uses heuristic approach to parse googleapis commitish and other metadata from google-cloud-java
-# See generate-library-list.sh and https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1390 for details
+# Get googleapis committish used for specified release and assign to global GOOGLEAPIS_COMMITTISH.
+# For each library in google-cloud-java, parse information from ".repo-metadata.json"
+# and hermetic build config file: generation_config.yaml
 # Assumes current working directory is spring-cloud-generator
 #
 # $1 - Monorepo version tag (or committish)
 function generate_libraries_list(){
   monorepo_commitish=$1
+  git clone --depth=1 https://github.com/googleapis/google-cloud-java.git
+  pushd google-cloud-java || { echo "Failure: google-cloud-java folder does not exists."; exit 1; }
+  # read googleapis committish used in hermetic build
+  GOOGLEAPIS_COMMITTISH=$(yq -r ".googleapis_commitish" generation_config.yaml)
```

Comment:
Is it still possible to keep the library level `googleapis_committish`? We do support it a different `googleapis_committish` at level level in `generation_config.yaml`. Vast majority of time every library in google-cloud-java should be generated with the same committish, but we did have a use case during last freeze that `java-language` wanted to release a new version, so in that case `java-language` would have a different committish.  It is a very rare case and could be addresses in a different PR.

Reply:
IUCC, this library level "googleapis_committish" is optional, if it's present, we should parse if from the libraries and it takes precedence of this repo level config?
If so, the current `generation_config.yaml` does not have libraries with this setting, and as you said this is a rare case. I would like to address it in a followup pr, so we don't block the OwlBot.yaml change for the release cycle.